packages/panzer/disc-fe/src/Panzer_BasisDescriptor.cpp

```diff
@@ -42,11 +42,53 @@
 
 #include "Panzer_BasisDescriptor.hpp"
 
+#include "Phalanx_KokkosDeviceTypes.hpp"
+
 #include "Panzer_HashUtils.hpp"
+#include "Panzer_IntrepidBasisFactory.hpp"
+#include "Panzer_PointGenerator.hpp"   // includes Kokkos::DynRankView
 
 namespace panzer
 {
 
+// Anonymous namespace that holds the coordinate generator,
+// this hides any details about the generator from an external
```

Comment:
I'm burying the implementation of the BasisCoordsGenerator in an anonymous namespace in the cpp file. This hides any details we about Kokkos we might need later.